pkg/antctl/raw/supportbundle/command.go

```diff
@@ -641,9 +656,184 @@ func processResults(resultMap map[string]error, dir string) error {
+
+		if controllerInfo != nil {
+			data, err := yaml.Marshal(controllerInfo)
+			if err != nil {
+				errors = append(errors, err)
+			} else {
+				err = afero.WriteFile(defaultFS, filepath.Join(tmpDir, "controllerinfo"), data, 0644)
+				errors = append(errors, err)
+			}
+		}
+
+		err = downloadPodLogs(k8sClient, "controller", pod.Namespace, pod.Name, []string{"antrea-controller"}, dir, tmpDir)
+		errors = append(errors, err)
+	}
+	return utilerror.NewAggregate(errors)
+}
+
+func downloadAgentBundleFromKubernetes(antreaClientset antrea.Interface, k8sClient kubernetes.Interface, failedNodes []string, dir string) error {
+	var errors []error
+	agentInfoList, err := antreaClientset.CrdV1beta1().AntreaAgentInfos().List(context.TODO(), metav1.ListOptions{ResourceVersion: "0"})
+	if err != nil {
+		// in case when there are no agentInfo exist, but there are still pods running. we can ignore this error
+		// and get pod logs
+		errors = append(errors, err)
+	}
+
+	agentInfoMap := map[string]v1beta1.AntreaAgentInfo{}
+	for _, agentInfo := range agentInfoList.Items {
+		agentInfoMap[agentInfo.Name] = agentInfo
+	}
+
+	pods, err := k8sClient.CoreV1().Pods("kube-system").List(context.TODO(), metav1.ListOptions{
+		ResourceVersion: "0",
+		LabelSelector:   "app=antrea,component=antrea-agent",
+	})
+	if err != nil {
+		return err
+	}
+
+	failedNodesSet := sets.NewString(failedNodes...)
+
+	for _, pod := range pods.Items {
+
+		tmpDir, err := afero.TempDir(defaultFS, "", "bundle_tmp_")
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		defer defaultFS.RemoveAll(tmpDir)
+
+		if !failedNodesSet.Has(pod.Spec.NodeName) {
+			continue
+		}
+
+		if agentInfo, ok := agentInfoMap[pod.Spec.NodeName]; ok {
+			data, err := yaml.Marshal(agentInfo)
+			if err != nil {
+				errors = append(errors, err)
+			}
+			err = afero.WriteFile(defaultFS, filepath.Join(tmpDir, "agentinfo"), data, 0644)
+			errors = append(errors, err)
+		}
+
+		var containerNames []string
+		for _, container := range pod.Spec.Containers {
+			containerNames = append(containerNames, container.Name)
+		}
+		for _, container := range pod.Spec.InitContainers {
+			containerNames = append(containerNames, container.Name)
+		}
+
+		err = downloadPodLogs(k8sClient, "agent_"+pod.Spec.NodeName, pod.Namespace, pod.Name, containerNames, dir, tmpDir)
+		errors = append(errors, err)
+
+	}
+	return utilerror.NewAggregate(errors)
+
+}
+
+func downloadPodLogs(k8sClient kubernetes.Interface, comp string, namespace string, podName string, containers []string, dir string, tmpDir string) error {
+	var errors []error
+
+	for _, containerName := range containers {
+		containerDirName := containerName
+		if strings.HasPrefix(containerName, "antrea-") {
+			containerDirName = strings.ReplaceAll(containerName, "antrea-", "")
+		}
+
+		podLogDir := filepath.Join(tmpDir, "logs", containerDirName)
+		err := os.MkdirAll(podLogDir, 0755)
+		if err != nil {
+			return err
```

Comment:
why it sometimes uses `return` without handling next container, sometimes uses `continue`?